### DISCUSSION TITLE
How can i Call a Dialog from a cshtml razor file?

### DISCUSSION THREAD

**Discussion by L30P1N0** [serenity-is/Serenity#7292] [Help]
The question is simple: in older versions of Serenity .NET Core, I could call dialogs from a .cshtml file simply by doing this:

```
jQuery(function () {
    var dlg = new TareasDialogReadOnly();
    //var dlg = new CheckList.Tareas.TareasDialog();
    dlg.MeLlaman = 'MANTEN';
    dlg.init();
    dlg.loadByIdAndOpenDialog(info.event.id);
    calendar.refetchEvents();
});
```

Now, it says that CTAApp is undefined. Does anyone have a solution for calling dialogs from my .cshtml file? I'm porting an old application to a newer version of the framework.

Any ideas?

**Comment by VictorTomaili** [comment]
Do you use ESM files? then the better way to use page.ts files. Don't write javascript code in your cshtml page. It possible import entry points with modular script tag, but dialogs are not an entry point. So importing them is not possible. You need to use *page.ts files to manipulate the page.
